client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/presentation/SessionActivity.java

```diff
@@ -1092,6 +1103,23 @@ public void onTouchPointerResetScrollZoom() {
         scrollView.scrollTo(0, 0);
     }
 
+    @Override
+    public boolean onGenericMotionEvent(MotionEvent e) {
+        super.onGenericMotionEvent(e);
+        switch (e.getAction()) {
+            case MotionEvent.ACTION_SCROLL:
+                final float vScroll = e.getAxisValue(MotionEvent.AXIS_VSCROLL);
```

Comment:
Optional, but may be a good idea to add horizontal mouse wheel too.

Reply:
Usually it is implemented on the side of OS. Client just handle events from mouse, and most of mouses have 1 scroll wheel

Reply:
Well, android usually has a touch screen, so it has a horizontal 'wheel' ;)

Reply:
This function handle actions from hardware mouse. But if it is optional, maybe consider it in further updates?